zcash_client_backend/src/data_api/chain.rs

```diff
@@ -140,6 +140,8 @@ pub trait BlockSource {
 /// - `block_source` Source of compact blocks
 /// - `from_tip` Height & hash of last validated block; if no validation has previously
 ///    been performed, this will begin scanning from `sapling_activation_height - 1`
+/// - `limit` amount of blocks that will be fetched from `block_source` and attempt to 
+///    be validated
 ///
 /// Returns:
 /// - `Ok(())` if the combined chain is valid.
```

Comment:
This case needs more explanation given the presence of `limit` (as the chain is not valid any more, but only "valid up to the given limit height").